packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -3631,6 +3632,62 @@ void main() {
     expect(tester.takeException(), isNull);
   });
 
+  testWidgets('NavigationRail labels shall not overflow if longer texts provided - extended', (WidgetTester tester) async {
+
+    // The navigation rail has a narrow width constraint. The text should wrap.
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(useMaterial3: true),
```

Comment:
Consider removing `theme: ThemeData(useMaterial3: true),`
This is no more required (M3 is the default now).

Reply:
Removed.